extensions/azurePublish/src/services/blobStorageService.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { StorageManagementClient } from '@azure/arm-storage';
+import { TokenCredentials } from '@azure/ms-rest-js';
+
+import { throwNotImplementedError } from './throwNotImplementedError';
+
+const createBlobStorageService = (token: string, subscriptionId: string) => {
+  const tokenCredentials = new TokenCredentials(token);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const storageManagementClient = new StorageManagementClient(tokenCredentials, subscriptionId);
+
+  const checkNameAvailability = async () => {
+    throwNotImplementedError();
+  };
+
+  const create = async () => {
+    throwNotImplementedError();
+  };
+
+  const deleteMethod = async () => {
+    throwNotImplementedError();
+  };
+
+  /**
+   * Returns the properties for the specified storage account
+   */
+  const getProperties = async () => {
+    throwNotImplementedError();
+  };
+
+  const list = async () => {
+    throwNotImplementedError();
+  };
+
+  const listByResourceGroup = async () => {
```

Comment:
Consider if the list method should have an optional argument(s) for filtering rather than a listByResourceGroup method.  Are there other ways to filter or does Azure usually only provide the by resource group?